### PR TITLE
doc(volumes) - clarify manual how-to for RWX volume expansion and add…

### PR DESCRIPTION
**Pull request by james-munson**
… caveat.

#### Which issue(s) this PR fixes:

Issue https://github.com/longhorn/longhorn/issues/6216

#### What this PR does / why we need it:

Be more explicit about the steps to grow an RWX volume.  Note that it can't be applied to XFS fsType.

#### Special notes for your reviewer:

When approved, I will backport to 1.5 and 1.6 release docs as well.

### CODE REVIEW

File: content/docs/1.7.0/nodes-and-volumes/volumes/expansion.md [hunk 1]
```diff
@@ -125,10 +125,19 @@ If you cannot enable it but still prefer to do online expansion, you can:
...
     ```shell
     kubectl -n longhorn-system exec -it <the share manager pod> -- resize2fs /dev/longhorn/<volume name>
     ```
+
+##### Offline
+1. Detach the RWX volume by scaling the workload down to replicas=0.  Make sure the volume is fully detached before the next step.
+2. Expand the block device via PVC or UI as above.
+3. Scale the workload back up.  The reattached volume will have the expanded size.
+
+> **NOTE** This can only be done for volumes with `fsType: ext4`.  If it is attempted with a volume of type `xfs`, it may appear to be successful, but when the workload is next scaled up or reattached, the pods will hang in `ContainerCreating` and the logs will show an error attempting to mount the filesystem.
```
Comment (jillian-maroket) [inline content/docs/1.7.0/nodes-and-volumes/volumes/expansion.md:142]:
```suggestion
> **Important**: 
> Manual expansion is only possible for volumes with `fsType: ext4`. Attempts to manually expand volumes with `fsType: xfs` may initially appear to be successful, but issues occur when the workload is scaled up and the volume is reattached. In particular, the pods will become stuck in the `ContainerCreating` state, and the logs will show an error message related to attempts to mount the filesystem.
```

**Review by jillian-maroket — COMMENTED** [review]
Let me know if you have concerns about the changes.

**Review by ejweber — COMMENTED** [review]
I think we should discuss https://github.com/longhorn/longhorn/issues/6216#issuecomment-2220812095 before moving forward with this one.

**Comment by james-munson** [comment]
When okayed, I will backport this to all current releases.

File: content/docs/1.7.0/nodes-and-volumes/volumes/expansion.md [hunk 2]
```diff
@@ -125,10 +125,27 @@ If you cannot enable it but still prefer to do online expansion, you can:
...
+##### Online
 1. Expand the block device of the RWX volume via PVC or UI.
-2. Figure out the share manager pod of the RWX volume then execute the filesystem expansion command. The share manager pod is typically named as `share-manager-<volume name>`.
+2. Identify the Share Manager pod of the RWX volume (typically named `share-manager-<volume name>`), and then run the filesystem expansion command in it.
     ```shell
     kubectl -n longhorn-system exec -it <the share manager pod> -- resize2fs /dev/longhorn/<volume name>
     ```
+
+> **Important**:  
+> Online expansion can only be done for volumes with `fsType: ext4`.  If it is attempted with a volume of type `xfs`with `xfs_growfs`, it may appear to be successful, but when the workload is next scaled up or reattached, the pods will hang in `ContainerCreating` and the logs will show an error attempting to mount the filesystem.
```
Comment (jillian-maroket) [inline content/docs/1.7.0/nodes-and-volumes/volumes/expansion.md:138]:
```suggestion
> Online expansion is possible only for `ext4` volumes. Attempts to manually expand `xfs` volumes with `xfs_growfs` may initially appear to be successful, but issues occur when the workload is scaled up and the volume is reattached. In particular, the pods become stuck in the `ContainerCreating` state, and the logs show an error message about attempts to mount the filesystem.
```

File: content/docs/1.7.0/nodes-and-volumes/volumes/expansion.md [hunk 3]
```diff
@@ -125,10 +125,27 @@ If you cannot enable it but still prefer to do online expansion, you can:
...
+
+##### Offline
+1. Detach the RWX volume by scaling the workload down to `replicas=0`.  
+2. Make sure the volume is fully detached.  After the scale command returns, check it with
```
Comment (jillian-maroket) [inline content/docs/1.7.0/nodes-and-volumes/volumes/expansion.md:142]:
```suggestion
##### Offline

1. Detach the RWX volume by scaling down the workload to `replicas=0`. Ensure that the volume is fully detached.

1. After the scale command returns, run the following command and verify that the state is `detached`.
```

File: content/docs/1.7.0/nodes-and-volumes/volumes/expansion.md [hunk 4]
```diff
@@ -125,10 +125,27 @@ If you cannot enable it but still prefer to do online expansion, you can:
...
+##### Offline
+1. Detach the RWX volume by scaling the workload down to `replicas=0`.  
+2. Make sure the volume is fully detached.  After the scale command returns, check it with 
+    ```shell
+    kubectl -n longhorn-system get volume <volume-name>
+    ```
+    and verify that the state is `detached`.  That may take a little time after the scale-down.
+3. Expand the block device using either the PVC or Longhorn UI.
+4. Scale up the workload.  
+The reattached volume will have the expanded size.
```
Comment (jillian-maroket) [inline content/docs/1.7.0/nodes-and-volumes/volumes/expansion.md:149]:
```suggestion
1. Expand the block device using either the PVC or the Longhorn UI.

1. Scale up the workload.

The reattached volume will have the expanded size.
```

**Review by jillian-maroket — COMMENTED** [review]
I reviewed the modified text.

**Comment by james-munson** [comment]
Branch history has gone astray.  Closing this PR and opening a new one.

**Pull request closed without merging**